src/blueapi/config.py

```diff
@@ -41,9 +41,22 @@ class StompConfig(BaseModel):
 
 
 class DataWritingConfig(BlueapiBaseModel):
+    """Configuration for where to write output files for ophyd-async devices.
+
+    Args:
+        visit_service_url (Optional[str]): the /api endpoint of a visit_service
+        beamline (str): beamline identifier to use in constructing the name of files
+        root (Optional[str]): root to use when constructing StreamResources.
+            When visits are part of the response from the visit_service,
+            this will be deprecated and replaced with a method that constructs
+            it (e.g. return f"/{year}/{visit_id})
+    """
+
+    beamline: str = "example"
```

Comment:
Very small nitpick but I don't like having the word "beamline" in blueapi, encourages keeping it generically deployable (e.g. in a CryoEM or a small lab setup)

Reply:
prefix=f"{self._beamline}-{collection_id}"

Is already so Diamond specific whether it's _beamline or_data_group_name....

And when the pre-processor is moved to dodal surely it's being more Diamond specific there?

I think DirectoryProvider should define an abstract "generate_metadata" and just call that, dodal implementation just calls its own get_directory_provider, and then any implementations (like how BlueAPI is acting prior to moving pre-processor to Dodal) is creating and setting the implementation of Directory provider.